src/components/matrix/column-tree-headers.tsx

```diff
@@ -0,0 +1,592 @@
+    }
+  }, [searchedColID]);
+
+  /**
+   * Handle toggle in tree view
+   */
+  const handleToggle = (event: React.SyntheticEvent, nodeIds: string[]) => {
+    setExpanded(nodeIds);
+  };
+
+  return (
+    <SharedColumnHeaders {...props}>
+      <div
+        className='grid-column-headers'
+        style={columnTreeHeadersStyles}
+        onScroll={props.onScroll}
+        ref={ref}
+      >
+        <div
+        style={{
+          transformOrigin: 'top left',
+          transform: 'rotate(-90deg)',
+        }}>
+
+          <TreeView
+            aria-label='rich object'
+            defaultExpanded={['root']}
+            defaultCollapseIcon={<MemoizedMinusSquare cellWidth={props.cellWidth} iconSize={iconSize} />}
+            defaultExpandIcon={<MemoizedPlusSquare cellWidth={props.cellWidth} iconSize={iconSize} />}
+            defaultEndIcon={<MemoizedCloseSquare cellWidth={props.cellWidth} />}
+            expanded={expanded}
+            onNodeToggle={handleToggle}
+            ref={divRef}
+            style={{
+              // height: 'fit-content',
+              // width: 'fit-content',
+              position: 'absolute',
+              /**
+               * For vertical scrolling function of the horizontal tree, we only want to scroll up. But after using
+               * transform, there is a blank space at the bottom of the tree. To eliminate the space, we set 
+               * transformOrigin to 'top left'. Then the initial y position of the tree does not focus on the tree 
+               * entries but blanks. So we set 'left' attribute to it and use a variable to memorize the position
+               * of the tree entries, then update it whenever scroll the tree vertically.
+               */
+              // left: -scrollTreeYIniPos, // Adjust the left value based on the desired position
+              left: -scrollableHeight, // Adjust the left value based on the desired position
+              /**
+               * For the highlight alignment issue, we're adding empty cells and headers in the falt column headers.
+               * But Mui is ignoring the empty tree elements, and we use 'transform' for the tree in column header.
+               * So we add a paddingBottom attribute as the style of the tree manually. 
+               */
+              paddingBottom: props.cellWidth + 15,
+            }}
+            sx={{ overflow: 'clip' }}
+          >
+            <MemoizedRenderTree
+              nodes={props.treeNodes}
+              data={props.itemData}
+              cellWidth={props.cellWidth}
+              treeNodesMap={props.treeNodesMap}
+              isScrolling={isScrolling}
+              scrollableHeight={scrollableHeight}
+            />
+          </TreeView>
+        </div>
+      </div>
+    </SharedColumnHeaders>
+  );
+};
+
+
+/**
+ * Create a component that customizes the minus square icon for tree view and then memorize it 
+ */
+type MemoizedIconSquareProps = {
```

Comment:
These icon components are identical between column and row. Create a `tree-button.tsx` file under `src/components` and move these two in there.